libc-bottom-half/clocks/clock.c

```diff
@@ -1,13 +1,43 @@
 #define _WASI_EMULATED_PROCESS_CLOCKS
 #include <time.h>
+#ifdef __wasilibc_use_wasip2
+#include <wasi/wasip2.h>
+#else
 #include <wasi/api.h>
+#endif
 #include <common/time.h>
 
 _Static_assert(
     CLOCKS_PER_SEC == NSEC_PER_SEC,
     "This implementation assumes that `clock` is in nanoseconds"
 );
 
+#ifdef __wasilibc_use_wasip2
+// Snapshot of the monotonic clock at the start of the program.
+static wall_clock_datetime_t start;
+
+// Use a priority of 10 to run fairly early in the implementation-reserved
+// constructor priority range.
+__attribute__((constructor(10)))
+static void init(void) {
+    wall_clock_now(&start);
```

Comment:
In wasip2, the [wall clock](https://github.com/WebAssembly/wasi-clocks/blob/main/wit/wall-clock.wit) is a distinct interface from the [monotonic clock](https://github.com/WebAssembly/wasi-clocks/blob/main/wit/monotonic-clock.wit), and I believe that all of this code should be using the monotonic clock instead of the wall clock.